### PR TITLE
replace @/generated shim with global variable

### PR DESCRIPTION
**Pull request by kyscott18**
Addresses https://t.me/ponder_sh/7662

Not certain this will fix it because the virtual module method was working locally as well.

### CODE REVIEW

File: packages/core/src/_test/setup.ts [hunk 1]
```diff
@@ -137,11 +137,8 @@ const defaultDatabaseServiceSetup: DatabaseServiceSetup = {
   indexing: "historical",
 };
 
-vi.mock("@/generated", async () => {
-  return {
-    instanceId: "1234",
-  };
-});
+// @ts-ignore
+globalThis.INSTANCE_ID = "1234";
```
Comment (0xOlias) [inline packages/core/src/_test/setup.ts:141]: I think we should use something like `__PONDER_INSTANCE_ID` to reduce the chance of a collision. See [Next.js example](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/setup-http-agent-env.ts).

File: packages/core/src/drizzle/index.ts [hunk 2]
```diff
@@ -225,7 +219,8 @@ export const onchainTable = <
   extra: extra;
   dialect: "pg";
 }> => {
-  if (instanceId === undefined) {
+  // @ts-ignore
+  if (globalThis.INSTANCE_ID === undefined) {
```
Comment (0xOlias) [inline packages/core/src/drizzle/index.ts:223]: It should be safe to do `const instanceId = globalThis.INSTANCE_ID` in the file scope, right?

File: packages/core/src/drizzle/index.ts [hunk 3]
```diff
@@ -225,6 +219,8 @@ export const onchainTable = <
   extra: extra;
   dialect: "pg";
 }> => {
+  // @ts-ignore
+  const instanceId: string | undefined = globalThis.__PONDER_INSTANCE_ID;
```
Comment (0xOlias) [inline packages/core/src/drizzle/index.ts:223]: Could add a `declare global` somewhere in `packages/core` to annotate this as `string | undefined`, would remove the need for `@ts-ignore` everywhere

https://stackoverflow.com/a/69429093

**Review by 0xOlias — APPROVED** [review]
LGTM, non-blocking nit

File: packages/core/src/drizzle/index.ts (hunk 3 above)
Comment (kyscott18) [reply packages/core/src/drizzle/index.ts:223]: Hmm this wasn't working for me

**Pull request merged**